lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -3900,6 +3900,63 @@ class DecomposeAtenClampMaxOp : public OpRewritePattern<AtenClampMaxOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenCosineSimilarityOp : public OpRewritePattern<AtenCosineSimilarityOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenCosineSimilarityOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value x1 = op.getX1();
+    Value x2 = op.getX2();
+    Value dim = op.getDim();
+ 
+    // Broadcast x1 and x2 to the same shape
+    SmallVector<int64_t> indexBroadcastShapeInt;
+    SmallVector<Value> indexBroadcastShapeValue;
+    computeBroadcastShape(rewriter, loc, x1, x2, indexBroadcastShapeInt, indexBroadcastShapeValue);
```

Comment:
nit: these lines look a bit long. Make sure to format code using `git clang-format`

Reply:
Yes I have already formatted the code

Reply:
Make sure you're specifying the right commit range when using `git clang-format`. For example, to format the last two commits, you would do `git clang-format HEAD~2`. This will make changes that you then need to commit.

In this case, these are the formatting changes I see when I run clang format locally:

```diff
@@ -188,7 +188,8 @@ static SmallVector<int64_t> computeDimsOrderForMoveDim(int64_t srcDimInt,
 }
 
 namespace {
-class DecomposeAtenCosineSimilarityOp : public OpRewritePattern<AtenCosineSimilarityOp> {
+class DecomposeAtenCosineSimilarityOp
+    : public OpRewritePattern<AtenCosineSimilarityOp> {
   using OpRewritePattern::OpRewritePattern;
   LogicalResult matchAndRewrite(AtenCosineSimilarityOp op,
                                 PatternRewriter &rewriter) const override {
@@ -196,46 +197,51 @@ class DecomposeAtenCosineSimilarityOp : public OpRewritePattern<AtenCosineSimila
     Value x1 = op.getX1();
     Value x2 = op.getX2();
     Value dim = op.getDim();
- 
+
     // Broadcast x1 and x2 to the same shape
     SmallVector<int64_t> indexBroadcastShapeInt;
     SmallVector<Value> indexBroadcastShapeValue;
-    computeBroadcastShape(rewriter, loc, x1, x2, indexBroadcastShapeInt, indexBroadcastShapeValue);
+    computeBroadcastShape(rewriter, loc, x1, x2, indexBroadcastShapeInt,
+                          indexBroadcastShapeValue);
     Type dtype = x1.getType().cast<BaseTensorType>().getOptionalDtype();
-    Type broadcastType =
-        ValueTensorType::get(op.getContext(), llvm::ArrayRef(indexBroadcastShapeInt), dtype);
+    Type broadcastType = ValueTensorType::get(
+        op.getContext(), llvm::ArrayRef(indexBroadcastShapeInt), dtype);
     Value indexBroadcastShapeTorchList = rewriter.create<PrimListConstructOp>(
         loc, Torch::ListType::get(Torch::IntType::get(op.getContext())),
         indexBroadcastShapeValue);
-    x1 = rewriter.create<AtenBroadcastToOp>(loc, broadcastType, x1, indexBroadcastShapeTorchList);
-    x2 = rewriter.create<AtenBroadcastToOp>(loc, broadcastType, x2, indexBroadcastShapeTorchList);
+    x1 = rewriter.create<AtenBroadcastToOp>(loc, broadcastType, x1,
+                                            indexBroadcastShapeTorchList);
+    x2 = rewriter.create<AtenBroadcastToOp>(loc, broadcastType, x2,
+                                            indexBroadcastShapeTorchList);
 
     // Compute the mul of A and B
-    Value dotProduct = rewriter.create<AtenMulTensorOp>(loc, broadcastType, x1, x2);
+    Value dotProduct =
+        rewriter.create<AtenMulTensorOp>(loc, broadcastType, x1, x2);
     Value cstFalse = rewriter.create<Torch::ConstantBoolOp>(loc, false);
     Value cstNone = rewriter.create<Torch::ConstantNoneOp>(loc);
     Value dimList = rewriter.create<PrimListConstructOp>(
-      loc, Torch::ListType::get(Torch::IntType::get(op->getContext())),
-      ValueRange{dim});
-    Value sumDotProduct =
-        rewriter.create<Torch::AtenSumDimIntListOp>(
-          loc, op.getType(), /*self=*/dotProduct, /*dim=*/dimList,
-          /*keepdim=*/cstFalse,
-          /*dtype=*/cstNone);
-          
+        loc, Torch::ListType::get(Torch::IntType::get(op->getContext())),
+        ValueRange{dim});
+    Value sumDotProduct = rewriter.create<Torch::AtenSumDimIntListOp>(
+        loc, op.getType(), /*self=*/dotProduct, /*dim=*/dimList,
+        /*keepdim=*/cstFalse,
+        /*dtype=*/cstNone);
+
     // Compute the norm of A and B
-    Value ord = rewriter.create<Torch::ConstantFloatOp>(loc, rewriter.getF64FloatAttr(2.0));
+    Value ord = rewriter.create<Torch::ConstantFloatOp>(
+        loc, rewriter.getF64FloatAttr(2.0));
     Value normA = rewriter.create<AtenLinalgVectorNormOp>(
         loc, op.getType(), x1, ord, dimList, /*keepdim=*/cstFalse,
         /*dtype=*/cstNone);
     Value normB = rewriter.create<AtenLinalgVectorNormOp>(
         loc, op.getType(), x2, ord, dimList, /*keepdim=*/cstFalse,
         /*dtype=*/cstNone);
-  
+
     // Compute the product of the norms
-    Value normProduct = rewriter.create<AtenMulTensorOp>(loc, op.getType(), normA, normB);
-    Value normProductClamp = rewriter.create<AtenClampOp>(loc, op.getType(), normProduct,
-                                             op.getEps(), /*max=*/cstNone);
+    Value normProduct =
+        rewriter.create<AtenMulTensorOp>(loc, op.getType(), normA, normB);
+    Value normProductClamp = rewriter.create<AtenClampOp>(
+        loc, op.getType(), normProduct, op.getEps(), /*max=*/cstNone);
     // Compute the final cosine similarity by division
     rewriter.replaceOpWithNewOp<AtenDivTensorOp>(
         op, op.getType(), sumDotProduct, normProductClamp);
```

Reply:
Thank you! Fixed